data_structures/binary_trees/words_alphabetical.c

```diff
@@ -0,0 +1,242 @@
+        currentNode->frequency = 1;
+        currentNode->left = NULL;
+        currentNode->right = NULL;
+        return currentNode;
+    }
+    int64_t compared = strcmp(word, currentNode->word);  ///< holds compare state
+
+    if (compared > 0)
+        currentNode->right = addWordToTree(word, currentNode->right);
+    else if (compared < 0)
+        currentNode->left = addWordToTree(word, currentNode->left);
+    else
+        currentNode->frequency++;
+
+    return currentNode;
+}
+
+/**
+ * @brief Writes contents of tree to another file alphabetically
+ * @param node pointer to current node
+ * @param file pointer to file
+ * @returns void
+ */
+void writeContentOfTreeToFile(struct Node *node, FILE *file)
+{
+    static uint64_t i = 1;  ///< for line numbering in write file
+    if (node != NULL)
+    {
+        writeContentOfTreeToFile(node->left, file);
+        fprintf(file, "%-5u \t %-9u \t %s \n", i++, node->frequency,
+                node->word);
+        writeContentOfTreeToFile(node->right, file);
+    }
+}
+
+/**
+ * @brief Stores word in memory
+ * @param word word to be stored in memory
+ * @returns a pointer to the newly allocated word
+ */
+char *getPointerToWord(char *word)
+{
+    char *string = (char *)malloc((strlen(word) + 1) *
+                                  sizeof(char));  ///< pointer to string
+    // + 1 is for the '\0' character
+    if (string != NULL)
+    {
+        strcpy(string, word);
+        return string;
+    }
+    endProgramAbruptly(
+        "\n A problem occurred while reserving memory for word \n");
+    return NULL;
+}
+
+/**
+ * @brief Reserves memory for new node
+ * @returns a pointer to the newly allocated node
+ */
+struct Node *allocateMemoryForNode()
+{
+    struct Node *node =
+        (struct Node *)malloc(sizeof(struct Node));  ///< pointer to node
+    if (node != NULL)
+        return node;
+    endProgramAbruptly(
+        "\n A problem occured while reserving memory for structure \n");
+    return NULL;
+}
+
+/**
+ * @brief Closes file after read or write
+ * @param file pointer to file to be closed
+ * @returns void
+ */
+void closeFile(FILE *file)
+{
+    if (fclose(file))
+        endProgramAbruptly(" \n A Problem Occurred while closing a file \n");
+}
+
+/**
+ * @brief Frees memory when program is terminating
+ * @param node pointer to current node
+ * @returns void
+ */
+void freeTreeMemory(struct Node *node)
+{
+    if (node != NULL)
+    {
+        freeTreeMemory(node->left);
+        freeTreeMemory(node->right);
+        free(node->word);  ///< freeing node->word because memory was allocated
+                           ///< using malloc
+        free(node);  ///< freeing node because memory was allocated using malloc
+    }
+}
+
+/**
+ * @brief Ends program due to an error
+ * @param errorMessage the error message to be printed
+ * @returns void
+ */
+void endProgramAbruptly(char *errorMessage)
+{
+    fprintf(stderr, "%s\n", errorMessage);
+    exit(EXIT_FAILURE);
+}
+
```

Comment:
Please add self-test implementations using `assert` (**see the [typical structure](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program) of a program as an example**).